x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.test.tsx

```diff
@@ -0,0 +1,629 @@
+    const ONE_SECOND = 1000;
+    jest.setTimeout(10 * ONE_SECOND);
+    HTMLElement.prototype.getBoundingClientRect = jest.fn(() => {
+      return {
+        width: 1000,
+        height: 1000,
+        x: 0,
+        y: 0,
+      } as DOMRect;
+    });
+
+    (useKibana as jest.Mock).mockImplementation(() => {
+      return {
+        services: kibanaServiceMock,
+      };
+    });
+
+    (useTimelineEvents as jest.Mock).mockImplementation(useTimelineEventsMock);
+
+    (useTimelineEventsDetails as jest.Mock).mockImplementation(() => [false, {}]);
+
+    (useSourcererDataView as jest.Mock).mockImplementation(useSourcererDataViewMocked);
+
+    (useIsExperimentalFeatureEnabled as jest.Mock).mockImplementation(
+      useIsExperimentalFeatureEnabledMock
+    );
+  });
+
+  describe('columns', () => {
+    it('should move column left correctly ', async () => {
+      const field = {
+        name: 'message',
+      };
+      const { container } = renderTestComponents();
+
+      await waitFor(() => {
+        expect(screen.getByTestId('discoverDocTable')).toBeVisible();
+      });
+      expect(container.querySelector(`[data-gridcell-column-id="${field.name}"]`)).toHaveAttribute(
+        'data-gridcell-column-index',
+        '3'
+      );
+
+      expect(
+        container.querySelector(`[data-gridcell-column-id="${field.name}"]`)
+      ).toBeInTheDocument();
+
+      fireEvent.click(
+        container.querySelector(
+          `[data-gridcell-column-id="${field.name}"] .euiDataGridHeaderCell__icon`
+        ) as HTMLElement
+      );
+
+      await waitFor(() => {
+        expect(screen.getByTitle('Move left')).toBeEnabled();
+      });
+
+      fireEvent.click(screen.getByTitle('Move left'));
+
+      await waitFor(() => {
+        const newColumns = customStore.getState().timeline.timelineById[TimelineId.test].columns;
+        expect(newColumns[0].id).toBe('message');
+      });
+    });
+
+    it('should move column right correctly ', async () => {
+      const field = {
+        name: 'message',
+      };
+      const { container } = renderTestComponents();
+
+      await waitFor(() => {
+        expect(screen.getByTestId('discoverDocTable')).toBeVisible();
+      });
+      expect(container.querySelector(`[data-gridcell-column-id="${field.name}"]`)).toHaveAttribute(
+        'data-gridcell-column-index',
+        '3'
+      );
+
+      expect(
+        container.querySelector(`[data-gridcell-column-id="${field.name}"]`)
+      ).toBeInTheDocument();
+
+      fireEvent.click(
+        container.querySelector(
+          `[data-gridcell-column-id="${field.name}"] .euiDataGridHeaderCell__icon`
+        ) as HTMLElement
+      );
+
+      await waitFor(() => {
+        expect(screen.getByTitle('Move right')).toBeEnabled();
+      });
+
+      fireEvent.click(screen.getByTitle('Move right'));
+
+      await waitFor(() => {
+        const newColumns = customStore.getState().timeline.timelineById[TimelineId.test].columns;
```

Comment:
I wonder if you can abstract `customStore.getState().timeline.timelineById[TimelineId.test]` away to a fn call and re-use throughout these tests?

Reply:
Done